### ISSUE TITLE
Hawk Alert accessibility issues

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#1899]
# Problem

Our Hawk Alert implementation has a couple problems with how we are inserting Hawk Alerts into Drupal's message system. First, a Siteimprove rule is be triggered for the `#drupal-live-announce` region: https://alfa.siteimprove.com/rules/sia-r54. The rule and subsequent testing seems to indicate that this might be getting triggered because we are inserting Hawk Alerts into the region with full HTML markup. Second, Hawk Alerts are getting wrapped in with a `role="alert"` which automatically causes them to be announced separately implicitly using `aria-live="assertive"`. Testing with VoiceOver indicated that this was resulting a broken and unpredictable user experience where all the `assertive` messages interrupt each other. This can result in only one Hawk Alert being read aloud in its entirety (if there are multiple).

## Additional information
Siteimprove rule: https://alfa.siteimprove.com/rules/sia-r54

## Steps to reproduce
- Visit https://admissions.stage.drupal.uiowa.edu/. If you don't see a couple Hawk Alerts, you may need to run the following command:
```
ddev drush @admissions.test config:set uiowa_alerts.settings hawk_alert.source https://emergency.stage.drupal.uiowa.edu/api/active
```
- Turn on VoiceOver (Cmd-F5)
- Reload the Admissions staging website.
- There are 3 messages that should be read out loud: The concatenated `#drupal-live-announce` version of the Hawk Alerts and both of those same messages separately via the `role="alert"` attribute. Obviously, its not the desired outcome that all 3 of those messages would be getting populated, but if they are all going to exist, they should all be ready out by VoiceOver. But you should hear that only one of them gets read aloud.
- Repeat a few times to hear how it changes which message gets read aloud.

# Proposed Solution

There are multiple ways we could go about fixing it, but this is the one that I think results in the best outcome and user experience:

```js
messages.add(alertContent, {
  id: id,
  // type: 'error', <-- remove this
  dismissible: false,
  announce: Drupal.t('Hawk Alert: @alert', { '@alert': item.attributes.alert }) // <-- add this
});
```
By removing the `type: 'error'`, we would no longer cause the individual Hawk Alerts to get wrapped in `role="alert"` when they are inserted into the status message region of the page (where they are fully marked up to display as HTML). This prevents the problem where the alerts are doubled in the live announcement system. By setting the `announce` property to a non-HTML string, they will get fed into the `#drupal-live-announce` region consistently with how other messages are populated and this will hopefully be enough to stop triggering the Siteimprove rule.

<details>
<summary>Original issue summary</summary>

## Problem

hawk alert needs aria-atomic="true" on role=alert element

## Proposed Solution

From https://github.com/uiowa/uiowa/issues/1899#issuecomment-654428792, we might not be able to do this without an update to Core's messenger API.
</details>

**Comment by richardbporter** [comment]
I don't think we can add it to that element as the markup is generated by the Drupal messages API. I'm not seeing an option to pass it in.

**Comment by richardbporter** [comment]
We can add this in our JS but I'm wondering if we actually need it. Alerts are added/removed in their entirety and not individual parts of it - time, description, etc. I actually don't see existing alerts being updated on the same page without a refresh happening.

**Comment by richardbporter** [comment]
I don't think we need this as alerts are not updated dynamically - they are only added or removed from a wrapper div. See https://default.dev.drupal.uiowa.edu/ as an example. 

@inspector508  can you confirm this?

**Comment by richardbporter** [comment]
Some more info: incoming alerts have `role=alert` which I think implies `aria-live`. In previous in testing, they were announced on creation. I'm not sure if removals are announced or not. We talked about maybe putting `aria-live` on the wrapper div with `aria-relevant=removals` to maybe cover that.

**Comment by mjoneill** [comment]
HawkAlert is on every page, and hence, the accessibility issue is getting identified on every SiteNow page. 

https://my2.siteimprove.com/Inspector/998437/A11Y/Page?pageId=35324141593&impmd=8624D70727DCE5205B022281DD27D66F&conf=a+aa+aria+si&issue=cantTell+failed&wcag=twopointone#/sia-r54/failed/bTdJMWpLRzI=;ZjNiZmI1Yzg=

(^^^ link will likely not be helpful once the issue fades.  details below)


Field input error is not announced in full
Issue
3.3.1: Error Identification
4.1.3: Status Messages
A user should be clear about the required input for each form field. If a field is not completed correctly, then an error message should be given. This tells the user that information is missing or that the information provided is in the wrong format.

Field inputs that are marked as aria-live="assertive" cause a screen reader to interrupt the user when an error occurs. Marking a field input as aria-atomic means that the screen reader announces the complete error message each time it occurs.

**Comment by joewhitsitt** [comment]
I think Rick's comment from https://github.com/uiowa/uiowa/issues/1899#issuecomment-654428792 is still relevant. The SiteImprove report suggests that we need to make the change to the drupal-announce div which comes from Drupal Core. There is an upstream issue related to this, but I am not sure if it is the correct approach and hasn't gained a lot of traction yet:

https://www.drupal.org/project/drupal/issues/3344778

**Comment by pyrello** [comment]
I think this issue needs to be re-titled? It appears to be incidental to the Hawk Alert because its populated into the `drupal-live-announce` div.

**Comment by briand44** [comment]
Grooming conversation... There is an upstream issue https://www.drupal.org/project/drupal/issues/3344778. We should try and test out the patch there and provide feedback to help that issue move forward. We may need SiteImprove to scan a non-production site to test here. Maybe the SiteImprove browser plugin reports the same issue as the scan and we can depend on that and avoid the time it would take to scan a dev environment site. We can use emergency.dev.drupal.uiowa.edu to create an alert to appear on a dev site for testing. When `drupal-live-announce` is `assertive`, Siteimprove reports this issues. In other areas where it is `polite`, Siteimprove is not throwing errors.

**Comment by joewhitsitt** [comment]
Asked a question in an Axe-Con session:

> An accessibility scanning tool has suggested that our aria-live=assertive region should also have aria-atomic=true. I think you made it sound like we should only implement it that way if the content should be re-announced in full. Is that correct?

> **Laura**
>You are correct. That sounds like a potential case of false positive scanning pattern where they assume all assertive or aria-live should also state atomic as true, and that's not necessarily true.
> **Aaron**
>+1 Laura. Answer is "it depends" on nature of content.

The presenter also tried to address this question and was surprised the tool was flagging it. They asked what tool and though I responded in chat, it was not provided to the presenter for further comment.

So as Rick mentioned earlier, these hawk alerts are added/removed on load. If anything else is added to these (e.g. drupal status messages for on-page form submits), I don't think we want the entire contents of the region to be re-read in full.

While we would be fixing for the success criterion (yay siteimprove scores), I don't think we would be improving the user experience.

**Comment by briand44** [comment]
blocked based on what we learned in AxeCon and we have an email out to TM to get his feedback.

**Comment by joewhitsitt** [comment]
Just recalled that TM had responded to our email with clarifying questions a while back about this and I failed to respond.

A series of videos related to aria-live regions came up in the https://practical-accessibility.today/ training, I tested this a bit more on admissions.stage where emergency.stage's endpoint with test alerts exists and found that the contents and priority are replaced by new announcements.

<img width="1287" height="736" alt="Image" src="https://github.com/user-attachments/assets/7c94c54e-db58-4d02-9b8f-99ac2ee3311b" />
<img width="1267" height="667" alt="Image" src="https://github.com/user-attachments/assets/f8ca2933-8c2d-4729-9a01-19f1242666a7" />

So instances where aria-atomic=true, it won't re-read the hawk alerts that were added on page load. What isn't clear from the D.o issue is whether aria-atomic is set to false for polite priority announcements.

**Comment by pyrello** [comment]
I just updated the issue summary. There were multiple approaches that Claude Code suggested for fixing the issues. For the proposed solution, I went with the one that seems like it would provide the best user experience.

**Comment by cory-skeers** [comment]
> For the follow-up issue, I thought we should engage with TM on this matter. Here is a draft email:

I'd split this in two for clarity:
> 
```
...We want to go beyond making SiteImprove happy with this and make sure the functionality works well and as expected for campus users so any thoughts from you or your contacts using the tools on a daily basis about how we should proceed would be greatly appreciated.
```
to

```
...We want to go beyond making SiteImprove happy with this and make sure the functionality works well and as expected for campus users. Any thoughts from you or your contacts using the tools on a daily basis about how we should proceed would be greatly appreciated.
```
otherwise I think this looks good.
